axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/projectgenerator/factory/ProjectGeneratorFactoryTask.java

```diff
@@ -85,18 +86,25 @@ public Project create(SaleOrder saleOrder) {
   public ActionViewBuilder fill(Project project, SaleOrder saleOrder, LocalDateTime startDate)
       throws AxelorException {
     List<ProjectTask> tasks = new ArrayList<>();
+    AtomicBoolean isAlreadyGenerated = new AtomicBoolean(false);
```

Comment:
please find a simpler way to make this check, using a boolean in a submethod is not a good practice.